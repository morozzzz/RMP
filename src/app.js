import React from 'react';
import { Switch, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import { hot } from 'react-hot-loader';
import MainPage from './containers/MainPage/MainPage';
import DetailPage from './containers/DetailPage/DetailPage';
import NotFoundPage from './components/NotFoundPage/NotFoundPage';
import './app.css';

// eslint-disable-next-line object-curly-newline
const App = ({ Router, location, context, store }) => (
    <Provider store={store}>
        <Router location={location} context={context}>
            <Switch>
                <Route path="/film/:id" component={DetailPage} />
                <Route path="/search" component={MainPage} />
                <Route exact path="/" component={MainPage} />
                <Route path="" component={NotFoundPage} />
            </Switch>
        </Router>
    </Provider>
);

export default hot(module)(App);
