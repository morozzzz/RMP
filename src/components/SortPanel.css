.sort-panel {
    display: flex;
    
}

.sort-panel * {
    margin: 0 0.5em;
}

.sort-button {
    background: none;
    border: none;
    outline: none;
}

.sort-button-selected {
    color: rgb(248, 129, 129);
}