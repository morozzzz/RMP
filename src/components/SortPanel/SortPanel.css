.sort-panel {
    display: flex;
    
}

.sort-panel * {
    margin: 0 0.5em;
}

.sort-panel__button {
    background: none;
    border: none;
    outline: none;
    font-weight: bold;
}

.sort-panel__button_selected {
    color: rgb(248, 129, 129);
}

.pointer {
    cursor: pointer;
}