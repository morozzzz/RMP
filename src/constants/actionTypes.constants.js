const SET_SEARCH_CRITERIA = 'SET_SEARCH_CRITERIA';
const UPDATE_SERACH_TEXT = 'UPDATE_SERACH_TEXT';
const MOVIES_FETCH_SUCCESS = 'MOVIES_FETCH_SUCCESS';
const UPDATE_SORT_TYPE = 'UPDATE_SORT_TYPE';
const SET_DETAILED_MOVIE = 'SET_DETAILED_MOVIE';
const RELATED_MOVIES_FETCH_SUCCESS = 'RELATED_MOVIES_FETCH_SUCCESS';

export {
    SET_SEARCH_CRITERIA,
    UPDATE_SERACH_TEXT,
    MOVIES_FETCH_SUCCESS,
    UPDATE_SORT_TYPE,
    SET_DETAILED_MOVIE,
    RELATED_MOVIES_FETCH_SUCCESS,
};
