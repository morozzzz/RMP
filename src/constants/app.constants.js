const CRITERIAS = {
    TITLE: 'title',
    GENRE: 'genres',
};

const SORT_TYPES = {
    RELEASE_DATE: 'release_date',
    RATING: 'vote_average',
};

const KEY_CODES = {
    ENTER: 13,
};

const SERVER_URL = 'https://reactjs-cdp.herokuapp.com';

const GET_MOVIES_URL = `${SERVER_URL}/movies`;

export {
    CRITERIAS,
    SORT_TYPES,
    KEY_CODES,
    GET_MOVIES_URL,
};
