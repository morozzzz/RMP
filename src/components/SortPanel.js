import React from 'react';
import './SortPanel.css';

class SortPanel extends React.Component {
    onSortButtonClick = (event) => {
        event.target && event.target.classList.toggle('sort-button-selected'); //eslint-disable-line
    };

    render() {
        return (
            <div className="sort-panel">
                <div>Sort by</div>
                <button onClick={this.onSortButtonClick} className="sort-button" type="button">release date</button>
                <button onClick={this.onSortButtonClick} className="sort-button" type="button">rating</button>
            </div>
        );
    }
}

export default SortPanel;
