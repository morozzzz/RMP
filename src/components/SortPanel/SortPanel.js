import React from 'react';
import { connect } from 'react-redux';
import { SORT_TYPES } from '../../constants/app.constants';
import { updateSortType } from '../../actions/search.actions';
import './SortPanel.css';

class SortPanel extends React.PureComponent {
    componentDidUpdate = () => {
        const { onSortUpdate } = this.props;

        onSortUpdate();
    }

    render() {
        const { sortBy, onSortClick } = this.props;

        return (
            <div className="sort-panel">
                <div>Sort by</div>
                <button id="release-button" value={SORT_TYPES.RELEASE_DATE} onClick={onSortClick} className={`${sortBy === SORT_TYPES.RELEASE_DATE && 'sort-panel__button_selected'} sort-panel__button`} type="button">release date</button>
                <button id="rating-button" value={SORT_TYPES.RATING} onClick={onSortClick} className={`${sortBy === SORT_TYPES.RATING && 'sort-panel__button_selected'} sort-panel__button`} type="button">rating</button>
            </div>
        );
    }
}

const mapStateToProps = state => (
    {
        sortBy: state.searchParams.sortBy,
    }
);

const mapDispatchToProps = dispatch => (
    {
        onSortClick: event => dispatch(updateSortType(event.target.value)),
    }
);

export default connect(mapStateToProps, mapDispatchToProps)(SortPanel);
