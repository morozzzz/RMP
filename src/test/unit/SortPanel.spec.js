import React from 'react';
import { shallow } from 'enzyme';
import SortPanel from '../../components/SortPanel/SortPanel';
import { SORT_TYPES } from '../../constants/app.constants';

describe('SearchPanel', () => {
    it('should be rendered correctly', () => {
        const wrapper = shallow(<SortPanel />);

        expect(wrapper).toMatchSnapshot();
    });

    it('componentDidUpdate() should invoke props.onSortUpdate() with state as args', () => {
        const onSortUpdateMock = jest.fn();

        const wrapper = shallow(<SortPanel onSortUpdate={onSortUpdateMock} />);
        const instance = wrapper.instance();

        const state = wrapper.state();

        instance.componentDidUpdate();

        expect(onSortUpdateMock).toHaveBeenCalledWith(state);
    });

    it('click on RATING button should set state.sortBy with SORT_TYPES.RATING', () => {
        const wrapper = shallow(<SortPanel onSortUpdate={() => {}} />);
        const ratingButton = wrapper.find('#rating-button');

        ratingButton.simulate('click');

        expect(wrapper.state().sortBy).toEqual(SORT_TYPES.RATING);
    });

    it('click on RELEASE DATE button should set state.sortBy with SORT_TYPES.RELEASE_DATE', () => {
        const wrapper = shallow(<SortPanel onSortUpdate={() => {}} />);
        const releaseButton = wrapper.find('#release-button');

        releaseButton.simulate('click');

        expect(wrapper.state().sortBy).toEqual(SORT_TYPES.RELEASE_DATE);
    });
});
