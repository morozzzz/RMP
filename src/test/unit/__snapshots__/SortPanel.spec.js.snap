// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPanel should be rendered correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
